cypress/integration/sidebar_test.js

```diff
@@ -0,0 +1,63 @@
+describe('Sidebar tests', () => {
+  beforeEach(() => {
+    cy.fixture('realm').as('realm');
+    cy.route('GET', '/appengine/health', '');
+    cy.route('GET', '/realmmanagement/health', '');
+    cy.route('GET', '/pairing/health', '');
+    cy.route('GET', '/flow/health', '');
+    cy.login();
+    cy.visit('/');
+  });
+
+  it('correctly renders sidebar elements', function () {
+    cy.get('.nav-col .nav').within(() => {
+      cy.get('.nav-brand')
+        .as('brand')
+        .next('.nav-link')
+        .as('home')
+        .next('.nav-item')
+        .next('.nav-link')
+        .as('interfaces')
+        .next('.nav-link')
+        .as('triggers')
+        .next('.nav-item')
+        .next('.nav-link')
+        .as('devices')
+        .next('.nav-link')
+        .as('groups')
+        .next('.nav-item')
+        .next('.nav-link')
+        .as('flows')
+        .next('.nav-link')
+        .as('pipelines')
+        .next('.nav-link')
+        .as('blocks')
+        .next('.nav-item')
+        .next('.nav-link')
+        .as('realmSettings')
+        .next('.nav-item')
+        .next('.nav-status')
+        .as('realmStatus')
+        .next('.nav-item')
+        .next('.nav-link')
+        .as('logout');
+      cy.get('@brand').should('have.attr', 'href', '/');
+      cy.get('@home').should('have.attr', 'href', '/').contains('Home');
+      cy.get('@interfaces').should('have.attr', 'href', '/interfaces').contains('Interfaces');
+      cy.get('@triggers').should('have.attr', 'href', '/triggers').contains('Triggers');
+      cy.get('@devices').should('have.attr', 'href', '/devices').contains('Devices');
+      cy.get('@groups').should('have.attr', 'href', '/groups').contains('Groups');
+      cy.get('@flows').should('have.attr', 'href', '/flows').contains('Flows');
+      cy.get('@pipelines').should('have.attr', 'href', '/pipelines').contains('Pipelines');
+      cy.get('@blocks').should('have.attr', 'href', '/blocks').contains('Blocks');
+      cy.get('@realmSettings').should('have.attr', 'href', '/settings').contains('Realm settings');
+      cy.get('@realmStatus').within(() => {
+        cy.contains('Realm');
+        cy.contains(this.realm.name);
+        cy.contains('API Status');
+        cy.contains('Up and running');
```

Comment:
We could also mock a failing API (return 5XX or 4XX error in 1 or more /health enpoints) and see that the message is "Degraded"